aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1549,9 +1673,13 @@ impl AptosVM {
                     Some(sender) => vec![sender, *execute_as],
                 };
 
+                let traversal_storage = TraversalStorage::new();
+                let mut traversal_context = TraversalContext::new(&traversal_storage);
+
                 self.validate_and_execute_script(
                     &mut tmp_session,
                     &mut UnmeteredGasMeter,
+                    &mut traversal_context,
```

Comment:
How does this work if the gas meter is unmetered? We still want to have the same limits on the number of dependencies, etc.?

Reply:
Gas will not be effective but the limits will still be enforced. Notice that system dependencies are excluded anyway